code/game/objects/effects/fluids.dm

```diff
@@ -47,10 +54,11 @@
 	ADD_ACTIVE_FLUID(src)
 
 /obj/effect/fluid/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	var/turf/simulated/T = loc
+	var/turf/simulated/T = get_turf(src)
 	if(istype(T))
+		T.zone?.fuel_objs -= src
```

Comment:
If `zone` is null, this will still runtime, so the `?` doesn't do what you want.